### PR TITLE
Metadata Editor: redirect to login URL if accessing the editor as anonymous user

### PR DESCRIPTION
**Pull request by jahow**
### Description

This PR adds a common route guard for all routes in the ME app.

Also removed an unused component.

### Architectural changes

none

### Quality Assurance Checklist

- [x] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [x] If **new logic** ⚙️ is introduced: unit tests were added
- [x] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

### CODE REVIEW

File: apps/metadata-editor/src/app/app.routes.ts [hunk 1]
```diff
@@ -10,100 +9,110 @@ import { NewRecordResolver } from './new-record.resolver'
...
-        path: 'my-org',
-        component: MyOrgUsersComponent,
-        pathMatch: 'prefix',
+        path: 'edit/:uuid',
+        component: EditPageComponent,
+        resolve: { record: EditRecordResolver },
       },
     ],
   },
-  { path: 'sign-in', component: SignInPageComponent },
```
Comment (jahow) [inline apps/metadata-editor/src/app/app.routes.ts:93]: this route was removed as it was obviously not needed anyway

File: apps/metadata-editor/src/app/app.routes.ts [hunk 2]
```diff
@@ -10,100 +9,110 @@ import { NewRecordResolver } from './new-record.resolver'
...
 
 export const appRoutes: Route[] = [
-  { path: '', redirectTo: 'catalog/search', pathMatch: 'prefix' },
   {
-    path: 'catalog',
-    component: DashboardPageComponent,
-    outlet: 'primary',
+    path: '',
+    canActivate: [AuthGuardService],
     children: [
```
Comment (jahow) [inline apps/metadata-editor/src/app/app.routes.ts:18]: Adding a root route (:thinking:) makes it easier to apply the same guard to all routes

File: apps/metadata-editor-e2e/src/e2e/dashboard.cy.ts [hunk 3]
```diff
@@ -314,4 +314,16 @@ describe('dashboard', () => {
...
   })
+
+  describe('when the user is not logged in', () => {
+    beforeEach(() => {
+      cy.visit('/catalog/search')
+    })
+    it('redirects to the login page', () => {
+      cy.url().should(
+        'include',
+        '/catalog.signin?redirect=http://localhost:4200/catalog/search'
```
Comment (LHBruneton-C2C) [inline apps/metadata-editor-e2e/src/e2e/dashboard.cy.ts:325]: Out of curiosity, where does this `redirect` value come from? The referer? It's often needed to be redirected after successful login, but I don't see it written in our code.

**Review by LHBruneton-C2C — APPROVED** [review]
Thanks for this improvement.
There is still one corner case when the authentication might fail, when the user is already wrongly connected (XSFR cookie set on the wrong path from the legacy UI), and the redirection is not activated then.
Otherwise, all good !

File: apps/metadata-editor-e2e/src/e2e/dashboard.cy.ts (hunk 3 above)
Comment (jahow) [reply apps/metadata-editor-e2e/src/e2e/dashboard.cy.ts:325]: it's computed here : https://github.com/geonetwork/geonetwork-ui/blob/main/libs/api/repository/src/lib/gn4/auth/auth.service.ts#L13

**Pull request merged**